inc/formanswer.class.php

```diff
@@ -127,8 +127,10 @@ public function canViewItem() {
 
       $groupUser = new Group_User();
       $groups = $groupUser->getUserGroups($currentUser);
-      if (in_array($this->fields['users_id_validator'], $groups)) {
-         return true;
+      foreach ($groups as $group) {
+         if ($this->fields['groups_id_validator'] == $group['id']) {
+            return true;
+         }
```

Comment:
bad implementation fixed : $groups was not an array of group IDs. It actually contains an array of array of group fields.